descwl_shear_sims/trivial_sim.py

```diff
@@ -250,3 +318,1004 @@ def make_wcs(*, scale, image_origin, world_origin, theta=None):
+    star_bleeds=False,
+):
+    """
+    make a grid sim with trivial pixel scale, fixed sized
+    exponentials and gaussian psf
+
+    Parameters
+    ----------
+    rng: numpy.random.RandomState
+        The random number generator
+    noise: float
+        Gaussian noise level
+    objlist: list
+        List of GSObj
+    shifts: array
+        Array with fields dx and dy, which are du, dv offsets
+        in sky coords.
+    dim: int
+        Dimension of image
+    psf: GSObject or PowerSpectrumPSF
+        the psf
+    psf_dim: int
+        Dimensions of psf image that will be drawn when psf func is called
+    dither: bool
+        If set to True, dither randomly by a pixel width
+    rotate: bool
+        If set to True, rotate the image randomly
+    cosmic_rays: bool
+        If True, put in cosmic rays
+    bad_columns: bool
+        If True, put in bad columns
+    """
+
+    dims = [dim]*2
+    cen = (np.array(dims)-1)/2
+
+    se_origin = galsim.PositionD(x=cen[1], y=cen[0])
+    if dither:
+        dither_range = 0.5
+        off = rng.uniform(low=-dither_range, high=dither_range, size=2)
+        offset = galsim.PositionD(x=off[0], y=off[1])
+        se_origin = se_origin + offset
+    else:
+        offset = None
+
+    if rotate:
+        theta = rng.uniform(low=0, high=2*np.pi)
+    else:
+        theta = None
+
+    se_wcs = make_wcs(
+        scale=SCALE,
+        theta=theta,
+        image_origin=se_origin,
+        world_origin=WORLD_ORIGIN,
+    )
+
+    convolved_objects, psf_gsobj = get_convolved_objects(
+        objlist=objlist,
+        psf=psf,
+        shifts=shifts,
+        offset=offset,
+        se_wcs=se_wcs,
+        se_origin=se_origin,
+    )
+
+    objects = galsim.Add(convolved_objects)
+
+    # everything gets shifted by the dither offset
+    image = objects.drawImage(
+        nx=dim,
+        ny=dim,
+        wcs=se_wcs,
+        offset=offset,
+    )
+
+    weight = image.copy()
+    weight.array[:, :] = 1.0/noise**2
+
+    image.array[:, :] += rng.normal(scale=noise, size=dims)
+    noise_image = image.copy()
+    noise_image.array[:, :] = rng.normal(scale=noise, size=dims)
+
+    bmask = get_mask(
+        image=image,
+        rng=rng,
+        cosmic_rays=cosmic_rays,
+        bad_columns=bad_columns,
+    )
+
+    if bright_objlist is not None:
+        timage = image.copy()
+
+        assert bright_shifts is not None
+        assert bright_mags is not None
+        assert mask_threshold is not None
+
+        bright_convolved_objects, _ = get_convolved_objects(
+            objlist=bright_objlist,
+            psf=psf,
+            shifts=bright_shifts,
+            offset=offset,
+            se_wcs=se_wcs,
+            se_origin=se_origin,
+        )
+        for i, obj in enumerate(bright_convolved_objects):
+
+            obj.drawImage(
+                image=timage,
+                offset=offset,
+                method='phot',
+                n_photons=10_000_000,
```

Comment:
why photons here?

Reply:
I think we need to compute the number of photons from the exposure time etc so that the noise is right?

Reply:
Using photons avoids fft artifacts on these huge bright stars.

We don't care about getting the noise right, we mask them entirely anyway